index.d.ts

```diff
@@ -88,9 +89,19 @@ declare namespace customPlugin {
   //
   // CUSTOM PLUGIN
   //
-  type BasicHandler<RequestType, ResponseType> = (this: DecoratedFastify, request: DecoratedRequest<RequestType>, reply: fastify.FastifyReply) => ResponseType
-  type Handler<RequestType> = BasicHandler<RequestType, void>
-  type AsyncHandler<RequestType> = BasicHandler<RequestType, Promise<any>>
+  type BasicHandler<
+    ServiceConfigType = NodeJS.Dict<string>,
+    RouteGeneric extends RouteGenericInterface = RouteGenericInterface,
```

Comment:
Those types are repeated often, it is not possible to assign to a specific type and reuse it?

Reply:
I've looked it up, but I can't find a way to extract a generic. It is not actually a type itself.
Do you know how to do it?